### PR TITLE
Add `possible_outbound_ip_addresses` to app service

### PR DESCRIPTION
**Pull request by tomasaschan**
This is useful for e.g. setting up firewall rules for Azure SQL - since an App Service can be rescheduled, it's possible that the `outbound_ip_addresses` field output at deploy-time does not catch all IP addresses you need to open up to be future-safe.

Note: I couldn't get the code to build and run tests on my Windows machine (neither from PowerShell, inspecting the makefile to see what was required, nor from WSL), so I've basically just looked at #700 to see what I need to do to support this. (I think this could have been made easier with a `.gitattributes` and an `.editorconfig` at the project root, that - at the very least - specify line endings to be what we expect.) Please feel free to request changes if I missed something 😄

### CODE REVIEW

**Comment by katbyte** [comment]
@tomasaschan,

I think this may be better to export as a `TypeList` rather than a comma separated field. However in that case we should probably also update `oubound_ip_addresses` to match?

**Comment by tomasaschan** [comment]
Yeah, I actually thought so to, but decided to follow #700 as closely as possible. The rationale in the discussion there was that it's easy enough to get that list using `split` if one needs it.

I think it's reasonable that they match, but I don't really care which way as long as they both go the same way :) But since changing `outbound_ip_addresses` would be a breaking change, while this isn't, I also figured this can be added with less friction, and a new PR that changes the return type of both attributes can be made in the 2.0 scope.

**Comment by katbyte** [comment]
Rational decision to me @tomasaschan, CSV for now it is!

File: azurerm/data_source_app_service.go [hunk 1]
```diff
@@ -183,6 +183,7 @@ func dataSourceArmAppServiceRead(d *schema.ResourceData, meta interface{}) error
 		d.Set("https_only", props.HTTPSOnly)
 		d.Set("default_site_hostname", props.DefaultHostName)
 		d.Set("outbound_ip_addresses", props.OutboundIPAddresses)
+		d.Set("possible_outbound_ip_addresses", props.PossibleOutboundIPAddresses)
```
Comment (metacpp) [inline azurerm/data_source_app_service.go]: It looks like that you missed the `possible_outbound_ip_addresses` in the schema ?

Comment (metacpp) [inline azurerm/data_source_app_service.go]: It looks like that you missed the `possible_outbound_ip_addresses` in the schema ?

File: website/docs/r/app_service.html.markdown [hunk 2]
```diff
@@ -205,6 +205,8 @@ The following attributes are exported:
 
 * `outbound_ip_addresses` - A comma separated list of outbound IP addresses - such as `52.23.25.3,52.143.43.12`
 
+* `possible_outbound_ip_addresses` - A comma separated list of outbound IP addresses - such as `52.23.25.3,52.143.43.12,52.143.43.17` - not all of which are necessarily in use. Superset of `outbound_ip_addresses`.
```
Comment (metacpp) [inline website/docs/r/app_service.html.markdown]: It looks like that you missed adding this to the documentation for the data source.

**Comment by katbyte** [comment]
For the test you could just add it to the basic one, like:

```go
func TestAccDataSourceAzureRMAppService_basic(t *testing.T) {
	dataSourceName := "data.azurerm_app_service.test"
	rInt := acctest.RandInt()
	location := testLocation()

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:  func() { testAccPreCheck(t) },
		Providers: testAccProviders,
		Steps: []resource.TestStep{
			{
				Config: testAccDataSourceAppService_basic(rInt, location),
				Check: resource.ComposeTestCheckFunc(
					resource.TestCheckResourceAttrSet(dataSourceName, "app_service_plan_id"),
					resource.TestCheckResourceAttrSet(dataSourceName, "possible_outbound_ip_addresses"),
					resource.TestCheckResourceAttrSet(dataSourceName, "outbound_ip_addresses"),
					resource.TestCheckResourceAttr(dataSourceName, "tags.%", "0"),
				),
			},
		},
	})
}
```

**Comment by tomasaschan** [comment]
@katbyte @metacpp Thanks for the review! I honestly hadn't thought about the data source getting the same properties, but of course that makes sense. I've added the new property to the schema for the data source as well.

I've also added a few assertions to the basic tests for both resource and data source, as well as documentation of these properties to the data source (it seems the reason I missed it in the first place is that the `outbound_ip_addresses` property was not documented either).

**Comment by tombuildsstuff** [comment]
Data Source tests pass:

screenshot 2018-12-20 at 13 32 11

**Comment by tombuildsstuff** [comment]
Resource tests pass:

screenshot 2018-12-20 at 14 12 18

**Comment by tomasaschan** [comment]
Woop woop! Thanks for taking this - will significantly simplify my deployments, once released 🎉

**Pull request merged**